src/service/ServiceManager.java

```diff
@@ -312,76 +332,99 @@ public int getRequestLimit(){
-	
+
 	/**
 	 * Label Services
 	 * */
-	
-	public List<Label> getLabels() throws IOException{
-		if(repoId != null){
+
+	public List<Label> getLabels() throws IOException {
+		if (repoId != null) {
 			return labelService.getLabels(repoId);
 		}
 		return new ArrayList<Label>();
 	}
-	
-	public Label createLabel(Label ghLabel) throws IOException{
-		if(repoId != null){
+
+	public Label createLabel(Label ghLabel) throws IOException {
+		if (repoId != null) {
 			return labelService.createLabel(repoId, ghLabel);
 		}
-		return null; //TODO:
+		return null; // TODO:
 	}
-	
-	public void deleteLabel(String label) throws IOException{
-		if(repoId != null){
+
+	public void deleteLabel(String label) throws IOException {
+		if (repoId != null) {
 			labelService.deleteLabel(repoId, label);
 		}
 	}
-	
-	public Label editLabel(Label label , String name) throws IOException{
-		if(repoId != null){
-			return (Label)labelService.editLabel(repoId, label, name);
+
+	public Label editLabel(Label label, String name) throws IOException {
+		if (repoId != null) {
+			return (Label) labelService.editLabel(repoId, label, name);
 		}
 		return null;
 	}
-	
+
 	/**
 	 * Milestone Services
 	 * */
-	public List<Milestone> getMilestones() throws IOException{
-		if(repoId != null){
+	public List<Milestone> getMilestones() throws IOException {
+		if (repoId != null) {
 			return milestoneService.getMilestones(repoId, STATE_ALL);
 		}
 		return new ArrayList<Milestone>();
 	}
-	
-	public Milestone createMilestone(Milestone milestone) throws IOException{
-		if(repoId != null){
+
+	public Milestone createMilestone(Milestone milestone) throws IOException {
+		if (repoId != null) {
 			return milestoneService.createMilestone(repoId, milestone);
 		}
 		return null;
 	}
-	
-	public void deleteMilestone(int milestoneNum) throws IOException{
-		if(repoId != null){
+
+	public void deleteMilestone(int milestoneNum) throws IOException {
+		if (repoId != null) {
 			milestoneService.deleteMilestone(repoId, milestoneNum);
 		}
 	}
-	
-	public Milestone editMilestone(Milestone milestone) throws IOException{
-		if(repoId != null){
-			return (Milestone)milestoneService.editMilestone(repoId, milestone);
+
+	public Milestone editMilestone(Milestone milestone) throws IOException {
+		if (repoId != null) {
+			return (Milestone) milestoneService
+					.editMilestone(repoId, milestone);
 		}
 		return null;
 	}
-	
-	
+
 	/**
 	 * Services for IssueEvent
 	 * */
-	public List<IssueEvent> getFeeds() throws IOException{
+	public List<IssueEvent> getFeeds() throws IOException {
 		ArrayList<IssueEvent> eventList = new ArrayList<IssueEvent>();
 		String user = getRepoOwner();
 		String repo = getRepoName();
+
+		// Pull Requests
+		int reqCount = 0;
+		List<PullRequest> requests = pullReqService.getPullRequests(
+				new SearchRepository(user, repo), IssueService.STATE_OPEN);
+		if (requests != null && !requests.isEmpty()) {
+			for (PullRequest request : requests) {
+				if (request != null) {
```

Comment:
This part is quite deeply nested. You could ensure that the list is non-null beforehand (it's okay if it's empty), then filter it for null elements before going into the `for` loop logic. That would eliminate 2 levels of nesting.